torch/ao/quantization/fx/_lower_to_native_backend.py

```diff
@@ -371,77 +380,61 @@ def _lower_weighted_ref_functional(
+        # We search for the pattern backwards, starting with the quantize node
+        # Quantize node args: (func, scale, zp, dtype)
+        if n.target != torch.quantize_per_tensor:
+            continue
+        q_node = n
+        (func_node, output_scale_node, output_zp_node, _) = q_node.args
+        if func_node.op != "call_function":
+            continue
+        # Handle cases where the functional op is wrapped in a ReLU
+        if func_node.target == F.relu:
+            relu_node = func_node
+            func_node = relu_node.args[0]
+        else:
+            relu_node = None
+        if func_node.target not in LOWER_FUNCTIONAL_MAP:
+            continue
+        if should_skip_lowering(func_node, qconfig_map):
+            continue
 
-                # Step 1: Replace quantized weights with packed weights, which will be folded later
-                quantized_weight = weight_dq_node.args[0]
-                weight_dtype = quantized_weight.args[-1]
-                if has_bias:
-                    bias = func_node.args[2]
-                else:
-                    bias = func_node.kwargs.get("bias", None)
-                prepack_args = (quantized_weight, bias)
-                if ref_func == F.linear:
-                    prepack_op = get_linear_prepack_op_for_dtype(weight_dtype)
-                else:
-                    raise ValueError("Lowering for functional currently only supports linear op")
-                insert_prepack_after = bias if has_bias else quantized_weight
-                with model.graph.inserting_after(insert_prepack_after):
-                    packed_weight = model.graph.create_node("call_function", prepack_op, prepack_args, {})
-
-                # Step 2: Replace reference pattern with the corresponding quantized op
-                func_node.args = (input_dq_node.args[0], packed_weight, output_scale_node, output_zp_node)
-                func_node.target = q_relu_func if is_relu else q_func
-                q_node.replace_all_uses_with(func_node)
-                output_zp_node.append(func_node)
-
-                # Clean up: Remove dequantize and quantize nodes and the old func node
-                for dqn in [input_dq_node, weight_dq_node]:
-                    dqn_input = dqn.args[0]
-                    dqn.replace_all_uses_with(dqn_input)
-                    model.graph.erase_node(dqn)
-                model.graph.erase_node(q_node)
-                if is_relu:
-                    model.graph.erase_node(relu_node)
+        # Step 1: Replace quantized weights with packed weights, which will be folded later
+        # Linear args: (dequantized inputs, dequantized weights[, bias])
+        # Conv args: (dequantized inputs, dequantized weights[, bias, stride, padding, dilation, groups])
+        (input_dq_node, weight_dq_node, *remaining_func_args) = func_node.args
+        quantized_weight = weight_dq_node.args[0]
```

Comment:
and quantize_weight is a quantize_per_tensor node?

Reply:
Actually it can be a `quantize_per_tensor` or `quantize_per_channel`, but yes I will add the check